tools/functional-tester/etcd-tester/checks.go

```diff
@@ -65,6 +67,93 @@ func (hc *hashChecker) Check() (err error) {
 	return nil
 }
 
+type leaseChecker struct {
+	leaseStressers []Stresser
+}
+
+func newLeaseChecker(leaseStressers []Stresser) Checker { return &leaseChecker{leaseStressers} }
+
+func (lc *leaseChecker) Check() error {
+	plog.Info("lease stresser invariant check...")
+	errc := make(chan error)
+	for _, ls := range lc.leaseStressers {
+		go func(s Stresser) { errc <- lc.invariantCheck(s) }(ls)
+	}
+	var errs []error
+	for i := 0; i < len(lc.leaseStressers); i++ {
+		if err := <-errc; err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) == 0 {
+		return nil
+	}
+	return fmt.Errorf("lease stresser encounters error: (%v)", errs[0])
+}
+
+func (lc *leaseChecker) invariantCheck(lStresser Stresser) error {
+	ls := lStresser.(*leaseStresser)
+	// check revoked leases
+	plog.Infof("revoked leases %v", ls.revokedLeases)
+	if err := leasesCheck(true, ls, ls.revokedLeases); err != nil {
```

Comment:
let's wrap around leasecheck. we should try to hide all bool args.

let's use checkLeasesExpired() {leasesCheck(true, ls, ls.revokedLeases)}